### PR TITLE
Extract ParseCIDR and modify attachDetachControllerName

### PR DESCRIPTION
**Pull request by NickrenREN**
**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/kube-controller-manager/app/controllermanager.go [hunk 1]
```diff
@@ -444,17 +444,18 @@ func StartControllers(controllers map[string]InitFunc, s *options.CMServer, root
...
 	cloud, err := cloudprovider.InitCloudProvider(s.CloudProvider, s.CloudConfigFile)
 	if err != nil {
 		return fmt.Errorf("cloud provider could not be initialized: %v", err)
 	}
 
+	_, clusterCIDR, err := net.ParseCIDR(s.ClusterCIDR)
+	if err != nil {
+		glog.Warningf("Unsuccessful parsing of cluster CIDR %v: %v", s.ClusterCIDR, err)
+	}
+
```
Comment (sjenning) [inline cmd/kube-controller-manager/app/controllermanager.go]: Why are we moving this block outside the `if ctx.IsControllerEnabled(nodeControllerName)` blocks?  Seems like we only need to do that in the case where the `nodeController` or `routeController` are enabled.

Comment (NickrenREN) [inline cmd/kube-controller-manager/app/controllermanager.go]: yes,but they are always set to be enabled and if so we do not need to parse that and print same err log twic

Comment (sjenning) [inline cmd/kube-controller-manager/app/controllermanager.go]: fair enough.  worst case, if someone has these controllers disabled, they get an extra warning

**Comment by NickrenREN** [comment]
Thanks @sjenning

**Comment by NickrenREN** [comment]
ping @lavalamp. Could I get the approve label ?

File: cmd/kube-controller-manager/app/controllermanager.go (hunk 1 above)
Comment (lavalamp) [inline cmd/kube-controller-manager/app/controllermanager.go]: I think I have a preference not to do this, also. Each of these if blocks should logically be its own function, and moving this up here makes them less self contained. And given that this happens once on startup, and that CIDR parsing isn't particularly computationally intensive, this isn't really an efficiency win, either.

Comment (NickrenREN) [inline cmd/kube-controller-manager/app/controllermanager.go]: Moved that back

**Comment by lavalamp** [comment]
/lgtm
/approve

(for future reference, I think these little spell check PRs are more trouble than they're worth.)

**Pull request merged**